src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1144,8 +1144,12 @@ async def validate_local_path(self, verrors, name, path):
                 raise
 
     @private
-    async def validate_share_name(self, name, schema_name, verrors, exist_ok=True):
-        if not exist_ok and await self.query([['name', 'C=', name]], {'select': ['name']}):
+    async def validate_share_name(self, name, schema_name, verrors, exist_ok=True, old=None):
```

Comment:
If `exist_ok` is `True` then this check (original and new versions) is a no-op.  Passing in `old` as below will have no impact.  Doesn't seem right.